### DISCUSSION TITLE
Disable Actor Resizing on Window Resizing

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvista#3304] [Q&A]
It seems by default, `pyvista` resizes all actors when the main window is resized:

https://user-images.githubusercontent.com/59346180/189797513-847672dc-2fd1-43f0-b7e4-2475a65108fa.mp4

1. Is there a way to disable this just for window resizing?
2. Is there a way to set specific actors not to resize on resizing a window or zooming/panning? (This can be useful for reference points like origins and axes)

**Comment by banesullivan** [comment]
Actors aren't "resized". When the window size changes, it is simply increasing the pixel size of the render view (increasing resolution)

You could add a custom event handler that zooms out on a window resize event, but getting the math right for the zoom factor will depend heavily on the scale/extent of the data being rendered in the scene

**Reply by adam-grant-hendry** [reply]
> Actors aren't "resized"...

Oh good to know.

> You could add a custom event handler that zooms out on a window resize event, but getting the math right for the zoom factor will depend heavily on the scale/extent of the data being rendered in the scene

Ah, thanks for that. That covers question 1.

Is question 2 possible? (I doubt it, but just curious, so I thought I'd ask. For 99% of what I need, attempting 1 will suffice)

**Reply by banesullivan** [reply]
> Is question 2 possible?

I think it's the wrong question.... The actors aren't being resized, the window resolution is changing, and thus question 2 seems no longer valid.

I suppose what you're getting at is what I mentioned about zooming out as the window size increases? I'm really not sure what you're trying to achieve here

**Reply by adam-grant-hendry** [reply]
> I suppose what you're getting at is what I mentioned about zooming out as the window size increases? I'm really not sure what you're trying to achieve here

Ya, agreed, 2 seems no longer valid. This is what I'm trying to achieve.

**Reply by banesullivan** [reply]
Okay, sorry I couldn't help much... if you manage to set up some callbacks with a zoom in/out on window resize, definitely share it here
